model/github/github_cache_entry.rb

```diff
@@ -5,14 +5,33 @@
 require "aws-sdk-s3"
 
 class GithubCacheEntry < Sequel::Model
+  plugin :instance_filters
+
   many_to_one :repository, key: :repository_id, class: :GithubRepository
 
   include ResourceMethods
 
+  dataset_module do
+    def destroy_where(...)
+      all do |entry|
+        entry.destroy_where(...)
+      end
+    end
+  end
```

Comment:
These 2 code blocks look like dark magic to me. Can you explain what do these do for educational purposes as a comment? I believe it won't be clear to anyone in the team either.

Reply:
Sure, I can add a comment.  Basically:

* `dataset_module` defines a module included in the model's dataset (this is used by a number of existing models already).
* `destroy_where(...)` defines a method (a dataset method in this case), that accepts any number of arguments.
* `all do |entry|` retrieves all entries in the dataset and iterates over all of them
* `entry.destroy_where(...)` calls the `destroy_where` method on the model instance, with the same arguments passed to the dataset method.
* Putting everything together, this is the same as the existing `Dataset#destroy` method, but it allows you to add an additional filter on the `DELETE` statement.

Reply:
I pushed another commit.  This drops the usage of `...` (generic argument forwarding), since the callers currently only pass a single argument.  I think that was likely the main source of confusion.  I also added a comment describing what the method does.